src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/TryHandler.java

```diff
@@ -42,10 +43,31 @@ public TryHandler(IndentationCheck indentCheck,
 
     @Override
     public IndentLevel getSuggestedChildIndent(AbstractExpressionHandler child) {
+
         if (child instanceof CatchHandler
             || child instanceof FinallyHandler) {
             return getIndent();
         }
+
+        final DetailAST resSpecAst = getMainAst().getFirstChild();
+        if (resSpecAst.getType() == TokenTypes.RESOURCE_SPECIFICATION) {
```

Comment:
This is a weird place to add the code.
This method is getting expected child indent, but you are doing validation on non-children elements and not adding to how the expected child indent should be calculated.

I think we should move this code to a new method that overrides `checkIndentation`, still calls the super, and then does your extra validation. This is how `MethodDefHandler` does it too.
This sound more reasonable?